disnake/flags.py

```diff
@@ -151,6 +132,29 @@ def __init__(self, **kwargs: bool):
                 raise TypeError(f"{key!r} is not a valid flag name.")
             setattr(self, key, value)
 
+    @classmethod
+    def __init_subclass__(cls, inverted: bool = False, no_fill_flags: bool = False):
+        # add a way to bypass filling flags, eg for ListBaseFlags.
+        if no_fill_flags:
+            return cls
+
+        # use the parent's current flags as a base if they exist
+        cls.VALID_FLAGS = getattr(cls, "VALID_FLAGS", {}).copy()
+
+        for name, value in cls.__dict__.items():
+            if isinstance(value, flag_value):
```

Comment:
```suggestion
            if isinstance(value, flag_value):
                if name in cls.VALID_FLAGS:
                    raise ValueError(f"Cannot redefine '{name}' flag")
```

Reply:
Hm, is this really necessary tho?

Reply:
Hmm, fair enough. Just thought this would be a good opportunity to add some sanity checks, but I suppose redefining a flag can be useful.